WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt

```diff
@@ -443,6 +463,42 @@ class MySiteViewModelTest : BaseUnitTest() {
         verify(currentAvatarSource).refresh()
     }
 
+    @Test
+    fun `given mySiteDashboardPhase2FeatureConfig is enabled, when refresh triggers, then all sources are refreshed`() {
+        whenever(mySiteDashboardPhase2FeatureConfig.isEnabled()).thenReturn(true)
+
+        viewModel.refresh()
+
+        allRefreshedMySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach { verify(it).refresh() }
+    }
+
+    @Test
+    fun `given mySiteDashboardPhase2FeatureConfig is disabled, when refresh triggers, then select sources refresh`() {
+        whenever(mySiteDashboardPhase2FeatureConfig.isEnabled()).thenReturn(false)
+
+        viewModel.refresh()
+
+        selectRefreshedMySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach { verify(it).refresh() }
+    }
+
+    @Test
+    fun `given mySiteDashboardPhase2FeatureConfig is enabled, when refreshing, then refresh indicator should show`() {
+        whenever(mySiteDashboardPhase2FeatureConfig.isEnabled()).thenReturn(true)
+
+        allRefreshedMySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach {
+            whenever(it.isRefreshing()).thenReturn(true)
+        }
+
+        assertThat(viewModel.shouldHideRefreshIndicator()).isFalse
+    }
+
+    @Test
+    fun `when mySiteDashboardPhase2FeatureConfig is disabled, then refresh indicator should hide`() {
```

Comment:
Suggestion (💡): Consider rewriting this test with a given/when/then structure, like the below:

```
    @Test
    fun `given mySiteDashboardPhase2FeatureConfig is disabled, when are sources refreshing, then refresh indicator should hide`() {
        whenever(mySiteDashboardPhase2FeatureConfig.isEnabled()).thenReturn(false)

        val result = viewModel.shouldHideRefreshIndicator()

        assertTrue(result)
    }
```

Reply:
Just me saving lines, I'll break into two. :(

Reply:
The test `when mySiteDashboardPhase2FeatureConfig is disabled, then refresh indicator should hide` is for when the  flag is disabled - meaning sources can't be refresh, because PTR isn't enabled.

Reply:
Nope, not the issue you ran into - that is all on SelectedSiteSource and that's a whole other discussion. lol.This test stands as written.